src/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -0,0 +1,119 @@
+import * as _ from "lodash";
+import * as classnames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+
+import Checkbox from "../Checkbox";
+import FormError from "../FormError";
+
+import "./CheckboxGroup.less";
+
+const OPTION_PROP_TYPE = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  checked: PropTypes.bool,
+  disabled: PropTypes.bool,
+  label: PropTypes.node.isRequired,
+  value: PropTypes.any,
+});
+
+const propTypes = {
+  className: PropTypes.string,
+  disabled: PropTypes.bool,
+  error: PropTypes.node,
+  label: PropTypes.node.isRequired,
+  onChange: PropTypes.func.isRequired,
+  options: PropTypes.arrayOf(OPTION_PROP_TYPE).isRequired,
+  title: PropTypes.node,
+};
+
+const cssClass = {
+  CHECKBOX: "CheckboxGroup--checkbox",
+  CHECKBOX_GLOBAL: "CheckboxGroup--checkbox--global",
+  CONTAINER: "CheckboxGroup",
+  LABEL: "CheckboxGroup--label",
+  SUB_OPTIONS: "CheckboxGroup--subOptions",
+};
+
+const LABEL_ID_PREFIX = "CheckboxGroup--label";
+let nextLabelIDSuffix = 0;
+const getNextLabelID = () => `${LABEL_ID_PREFIX}--${nextLabelIDSuffix++}`;
```

Comment:
Whoa! How does this work?

Reply:
oh yeah - we keep a shared reference to the next id and increment it here every time we use it, so that every new instance of the checkbox group gets a unique id for its label